src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -783,348 +179,163 @@ const std::vector<impl_desc_type>& FullyConnected::getDefaultImplPriority() {
-            addSupportedPrimitiveDescriptor(first_desc);
-    }
-}
+    factory->filter(key);
+    const auto descriptors = factory->preconfigureMemoryDescriptors(key);
 
-std::shared_ptr<MemoryDesc> FullyConnected::getSrcMemDesc(const dnnl::primitive_desc &prim_desc, size_t idx) const {
-    auto desc = idx > 0 ? prim_desc.weights_desc(idx - 1) : prim_desc.src_desc(idx);
-
-    if (getInputShapeAtPort(idx).getRank() != 2
-        // report original plain layout for weight since it needs to be reordered dynamically at runtime
-        || (idx == 1 && !useSparseWeights)) {
-        return std::make_shared<CpuBlockedMemoryDesc>(
-            DnnlExtensionUtils::DataTypeToElementType(desc.get_data_type()), getInputShapeAtPort(idx));
+    NodeConfig nodeConfig;
+    for (size_t i = 0; i < descInputNumbers(); i++) {
+        nodeConfig.inConfs.emplace_back(descriptors.src[i]);
     }
 
-    if (getInputShapeAtPort(idx).isDynamic()) {
-        return DnnlExtensionUtils::makeUndefinedDesc(desc, getInputShapeAtPort(idx));
+    for (size_t i = 0; i < descOutputNumbers(); i++) {
+        const int inPlace = canBeInPlace() ? 0 : -1;
+        nodeConfig.outConfs.emplace_back(descriptors.dst[i], BlockedMemoryDesc::FULL_MASK, inPlace);
     }
 
-    return DnnlExtensionUtils::makeDescriptor(desc);
+    supportedPrimitiveDescriptors.emplace_back(nodeConfig, impl_desc_type::undef);
 }
 
-std::shared_ptr<MemoryDesc> FullyConnected::getDstMemDesc(const dnnl::primitive_desc &prim_desc, size_t idx) const {
-    auto desc = prim_desc.dst_desc(idx);
-
-    if (getOutputShapeAtPort(idx).getRank() != 2) {
-        return std::make_shared<CpuBlockedMemoryDesc>(
-            DnnlExtensionUtils::DataTypeToElementType(desc.get_data_type()), getOutputShapeAtPort(idx));
-    }
-
-    if (getOutputShapeAtPort(idx).isDynamic()) {
-        return DnnlExtensionUtils::makeUndefinedDesc(desc, getOutputShapeAtPort(idx));
-    }
+void FullyConnected::createPrimitive() {
+    const auto& srcMemory = getParentEdgeAt(DATA_ID)->getMemoryPtr();
+    const auto& weiMemory = getParentEdgeAt(WEIGHTS_ID)->getMemoryPtr();
+    const auto& biaMemory = attrs.withBias ? getParentEdgeAt(BIAS_ID)->getMemoryPtr() : emptyMemory;
+    const auto& dstMemory = getChildEdgeAt(0)->getMemoryPtr();
+
+    memory[ARG_SRC] = srcMemory;
+    memory[ARG_WEI] = weiMemory;
+    memory[ARG_BIAS] = biaMemory;
+    memory[ARG_DST] = dstMemory;
+
+    descriptors.src.push_back(srcMemory->getDescPtr());
+    descriptors.src.push_back(weiMemory->getDescPtr());
+    attrs.withBias ? descriptors.src.push_back(biaMemory->getDescPtr())
+                   : descriptors.src.push_back(emptyMemory->getDescPtr());
+    descriptors.dst.push_back(dstMemory->getDescPtr());
+
+    const FCConfig key{descriptors, attrs, postOps};
+    // @todo should we preconfigure only for dynamic shapes?
+    // Since for static shapes primitive is created in scope of compile_model() anyway
+    factory->preconfigure(key, memory);
```

Comment:
Is this stage even mandatory? What happens if we carelessly don't call this method in `createPrimitive`?

Reply:
This method is necessary to prepack weights for some of the backends (i.e. MLAS and oneDNN on ARM)
"preconfigure" is just more generic name, in case executor needs to prepare more than just weights.

Reply:
So we just lose first inference performance. Got it.
A correct solution would be to instantiate a new object every new initialization step, but it's hardly feasible as the node itself is initialized via changing the internal state. So may be to reduce a probability of error, we can track the factory's initialization status via some internal state variable and throw upon attempt of calling `make` to uninitialized factory.